### ISSUE TITLE
getText() order of reading table is wrong

### ISSUE THREAD

**Issue by paddibr** [smalot/pdfparser#317]
Hi,

I have the following pdf file [2019-wk39wo.pdf](https://github.com/smalot/pdfparser/files/4890778/2019-wk39wo.pdf).

So if I mark the text by hand it does not mark all kind of text elements

![921a9c7f-b2a7-4946-8fc7-b76ad6555e67](https://user-images.githubusercontent.com/9463756/86921965-421add00-c12c-11ea-9ca1-4c6ff6c1ec42.jpg)

if I use the function getText() pdfparser is reading the text with the marked order. And not like the order of the table :(

As you can see here: the II and III is at the end of the array.

![array](https://user-images.githubusercontent.com/9463756/86922387-d422e580-c12c-11ea-99c0-326cfbd3a5fc.jpg)

Is there another solution to read the table??

**Comment by Connum** [comment]
I don't think this is really an issue with the library, but simply the way PDF works. The commands placing the text don't have to be in the order that the text visually appears on screen. What you should do to ensure the order of the parsed text: Use `getDataTm()`, then sort the text blocks into "rows" by stacking them into a nested associative array using the X and Y coordinates from the text matrix (Y sorted ascending for the "rows", that is the root array elements, then X ascending in each root element; you might need to account for slight differences in the coordinates).

Of course this could be wrapped in a handy function for the library to poivide this as a convenient feature out-of-the-box, but that would rather be an enhancement in my opinion. I might be able to provide this functionality soon, as I'm currently working on something where this would come in handy too.
